main.go

```diff
@@ -34,17 +34,30 @@ type BuildConfig struct {
 	debug      bool
 	printSizes string
 	initInterp bool
+	cFlags     []string
+	ldFlags    []string
 }
 
 // Helper function for Compiler object.
 func Compile(pkgName, outpath string, spec *TargetSpec, config *BuildConfig, action func(string) error) error {
 	if config.gc == "" && spec.GC != "" {
 		config.gc = spec.GC
 	}
+
+	// Append command line passed CFlags and LDFlags
+	if len(config.cFlags) > 0 && config.cFlags[0] != "" {
+		spec.CFlags = append(spec.CFlags, config.cFlags...)
+	}
+
+	if len(config.ldFlags) > 0 && config.ldFlags[0] != "" {
+		spec.LDFlags = append(spec.LDFlags, config.ldFlags...)
```

Comment:
I think you can avoid these complicated ifs when you check for an empty string below.
So instead of:

```go
&BuildConfig{
    // ...
    cFlags:     strings.Split(*cFlags, " "),
    // ...
}
```

You can add them this way:

```go
if *cFlags != "" {
    config.cFlags = strings.Split(*cFlags, " ")
}
```

Reply:
Done.